### ISSUE TITLE
Need help with externalized object

### ISSUE THREAD

**Issue by sagopaj** [mtanksl/amf.net#1]
Hello mtanksl,

I really appreciated your work on this project. Nice job!!

I need to automate an old program just like you but this program sends Externalized object to server which I didn't know the map of it.

Is there any way I create a custom class with externalizable interface and see what is inside the request which program is sending. 

I want to see inside of that externalized object that sended by program but don't know how to use [traitclass] or [traitmember]. I only got class name of externalized object from program but when I create a class in my code and give [trairclass]("name of the class from request send by program") I can't see anything inside that class. There is no members in it.

So please give me a hand about this. 

Thx 

Regards.

**Comment by mtanksl** [comment]
Hello sir,

I've added a new test file (ExternalizableSerializationDeserialization.cs) with two examples, one for serializing and deserializing a known externalizable class and other for deserializing an unknown externalizable class. Note that all you could see are the raw bytes, and you will have to figure it out what they mean. Quoting the documentation:

"Note that for U29O-traits-ext, after the class-name follows an indeterminable number of bytes as *(U8). This represents the completely custom serialization of "externalizable" types. The client and server have an agreement as to how to read in this information."
 
Good luck with your project, I can already feel your pain!
Best Regards

**Comment by sagopaj** [comment]
Hello again buddy,

First of all, thank you very much for your interest.

I wish such a project that solved an extremely difficult problem would be appreciated by many more people.

I have examined the examples you gave, but I am already tracking data traffic using Fiddler and Charles programs. So I can already see the byte data in the Externalized Object. But I do not know how to interpret this data, and how difficult it is to interpret it. If interpreting this data is not too difficult, I can study and learn. But it looks like I won't have to bother if it's too hard.

In this context, can I request something from you? If I send you the POST REQUEST's byte data of the AMF message I received from the Fiddler program, can you tell me whether this AMF message(I mean Externalized object in this amf message) can be simulated or is it impossible to do this task?

Here is the byte array of the complete AMF message I want to simulate

[byteArrayOfCompleteAmfMessage.txt](https://github.com/mtanksl/amf.net/files/6369150/byteArrayOfCompleteAmfMessage.txt)

And here is picture of Charles view of this Amf Message
![charlesViewOfThisAmfMessage](https://user-images.githubusercontent.com/83106124/115951613-4745ee80-a4ea-11eb-978a-39cc429eb71d.jpg)


With this Amf Message, I write "Hello" to a textbox in program and clicked "Save" button. And program send this Amf message to server. And save "Hello" text as a note into the database.

Thank you so much again. And apologize for taking your time.

Regards.

**Comment by mtanksl** [comment]
Hi!

I've updated the mtranksl.ActionMessageFormat.Viewer project with your data. Just hit F5 and you will see the AMF message parsed like so:

![unknown bytes in object viewer](https://user-images.githubusercontent.com/20389632/115968031-065ed180-a50c-11eb-96cc-a006c2675162.png)

Those are the bytes for the custom iexternizable. You will have to figure it out what they mean. I had to use an hex viewer to count and find out how many bytes for each iexternizable.

![hex viewer](https://user-images.githubusercontent.com/20389632/115968093-6eadb300-a50c-11eb-8007-4df0976598cd.png)

Have fun and best of luck!
